libraries/AP_RangeFinder/AP_RangeFinder_UAVCAN.cpp

```diff
@@ -0,0 +1,167 @@
+ */
+AP_RangeFinder_UAVCAN::AP_RangeFinder_UAVCAN(RangeFinder::RangeFinder_State &_state, AP_RangeFinder_Params &_params) :
+    AP_RangeFinder_Backend(_state, _params)
+{}
+
+void AP_RangeFinder_UAVCAN::subscribe_msgs(AP_UAVCAN* ap_uavcan)
+{
+    if (ap_uavcan == nullptr) {
+        return;
+    }
+
+    auto* node = ap_uavcan->get_node();
+
+    uavcan::Subscriber<uavcan::equipment::range_sensor::Measurement, MeasurementCb> *measurement_listener;
+    measurement_listener = new uavcan::Subscriber<uavcan::equipment::range_sensor::Measurement, MeasurementCb>(*node);
+    // Msg Handler
+    const int measurement_listener_res = measurement_listener->start(MeasurementCb(ap_uavcan, &handle_measurement));
+    if (measurement_listener_res < 0) {
+        AP_HAL::panic("UAVCAN RangeFinder subscriber start problem\n\r");
+        return;
+    }
+}
+
+AP_RangeFinder_UAVCAN* AP_RangeFinder_UAVCAN::get_uavcan_backend(AP_UAVCAN* ap_uavcan, uint8_t node_id, uint8_t address, bool create_new)
+{
+    if (ap_uavcan == nullptr) {
+        return nullptr;
+    }
+    AP_RangeFinder_UAVCAN* driver = nullptr;
+    for (uint8_t i = 0; i < RANGEFINDER_MAX_INSTANCES; i++) {
+        if (AP::rangefinder()->params[i].type == RangeFinder::RangeFinder_TYPE_UAVCAN &&
+            AP::rangefinder()->params[i].address == address) {
+            driver = (AP_RangeFinder_UAVCAN*)AP::rangefinder()->drivers[i];
+        }
+        //Double check if the driver was initialised as UAVCAN Type
+        if (driver != nullptr && (driver->_backend_type == RangeFinder::RangeFinder_TYPE_UAVCAN)) {
+            if (driver->_ap_uavcan == ap_uavcan && 
+                driver->_node_id == node_id) {
+                return driver;
+            }
+        }
+    }
+    
+    if (create_new) {
+        for (uint8_t i = 0; i < RANGEFINDER_MAX_INSTANCES; i++) {
+            if (AP::rangefinder()->params[i].type == RangeFinder::RangeFinder_TYPE_UAVCAN &&
+                AP::rangefinder()->params[i].address == address) {
+                if (AP::rangefinder()->drivers[i] != nullptr) {
+                    continue;
+                }
+                AP::rangefinder()->drivers[i] = new AP_RangeFinder_UAVCAN(AP::rangefinder()->state[i], AP::rangefinder()->params[i]);
+                driver = (AP_RangeFinder_UAVCAN*)AP::rangefinder()->drivers[i];
+                if (driver == nullptr) {
+                    break;
+                }
+                AP::rangefinder()->num_instances = MAX(i+1, AP::rangefinder()->num_instances);
+                if (driver->_ap_uavcan == nullptr) {
+                    driver->_ap_uavcan = ap_uavcan;
+                    driver->_node_id = node_id;
+                    break;
+                }
+            }
+        }
+    }
+
+    return driver;
+}
+
+void AP_RangeFinder_UAVCAN::update()
+{
+    WITH_SEMAPHORE(_sem);
+    if ((AP_HAL::millis() - _last_reading_ms) > 200) {
+        set_status(RangeFinder::RangeFinder_NoData);
+        state.distance_cm = 0;
```

Comment:
Why bother with this? We should be using the status to guard it, making this a throw away value. (And if isn't guarded this is also a problematic number)